worker/worker.go

```diff
@@ -392,6 +394,34 @@ func (w *worker) withHostV2(ctx context.Context, contractID types.FileContractID
 	})
 }
 
+func (w *worker) withRevision(ctx context.Context, contractID types.FileContractID, hostKey types.PublicKey, hostIP string, lockPriority int, lockDuration time.Duration, fn func(revision types.FileContractRevision) error) error {
```

Comment:
(nit) you introduce `hostKey` here but changed it to `hk` in other places. I'd probably keep one or the other.